src/dataset.rs

```diff
@@ -686,6 +682,10 @@ impl Dataset {
         };
 
         let c_layer = unsafe {
+            let c_srs = match options.srs {
+                Some(srs) => srs.to_c_hsrs(),
+                None => null_mut(),
+            };
```

Comment:
Since `to_c_hsrs` is now `unsafe`, moved from top of function to fall inside the same `unsafe` block as the call the the C API.